### DISCUSSION TITLE
Publishing Malware on GitHub

### DISCUSSION THREAD

**Discussion by nickcyberpunk** [community/community#58229] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi guys,
I developed a ransomware for testing/demonstration purposes as well as contributing to the security research community. 
The software simulates real C&C communication by fetching a remote RSA public key from my GitHub repository. The private key will also be posted on the repos to test both encryption and decryption. 

I'm planning on publishing the software, however I wanted to reach out to you to understand your policies when it comes to posting malware on your platform. I noticed many other users have published malicious software like info stealers, ransomware, etc. 

In my case, both encryption and decryption keys will be put online. However, it's trivial for someone with basic programming skills to use their personal key pair instead. This could potentially lead to someone using the software to do damage, given that the tool is not detected by some AV vendors and without the private keys the data is unrecoverable. 
  
Is the 'Disclaimer' section enough to stay on the safe side of things legally speaking?
I live in Italy and I did not find any documentation online specific to my country on this topic. 

Can you guys provide some advice on this topic? 

I'm trying to do the right thing and I just want to hear your point of view before anything else. 

Thanks for your help,

Cheers

**Comment by the-officialjosh** [comment]
I highly doubt there have strict policies against that. And because people do it doesn't guarantee your account might not be suspended if you do it. Better still reach out to GitHub support for more guidance on this

**Reply by nickcyberpunk** [reply]
Thanks for your feedback. 
Do you know how I can reach out to them? I tried support@github.com, but it does not work anymore.

**Comment by sbouillon** [comment]
I have yet to see any accounts get suspended even after posting vault7 source code, WannaCry, multitude of Ransomware-PoC and more/less destructive code. 

If you want to post it, then post it. If you would rather post it on some other platform and mirror it to GitHub then do so. 

As long as you’re not trying to exploit GitHub or break some law, I’m sure you’re fine.

**Comment by jmerc77** [comment]
recently, someone's repo with bulk malware got taken down even though it was intended for research. its risky. **I advise against it.** maybe you could provide via torrent. though I'm not sure, you may be able to only have the key on some site and the mag link. but still, not a good idea.

**Reply by sbouillon** [reply]
Are you able to point me to an actual repo that was taken down for this?

**Reply by jmerc77** [reply]
unfortunately not. but just found out that there were many. a lot from russia.

**Comment by dhrubasaha08** [comment]
Hello @nickcyberpunk,

While I appreciate your intentions and interest in contributing to the security research community, I strongly advise against sharing malware, including ransomware, on any public platform like GitHub.

Sharing such malicious software can have serious consequences, as it can potentially be misused by individuals with malicious intent. Furthermore, hosting encryption keys online can pose significant security risks, even if it is intended for demonstration purposes.

If you want to contribute to the security research community, I recommend finding alternative ways to share your findings and research, such as through private security communities, controlled environments, or by collaborating with trusted researchers in the field. It's important to prioritize ethical considerations and ensure that your actions align with legal regulations in your country.

I encourage you to consult with legal professionals who specialize in cybersecurity or related fields for specific advice regarding the legal implications in your country, Italy. They will be able to provide guidance based on the local laws and regulations that apply.

Thank you for reaching out, and I hope this information helps you in making the right decision.

Best regards

**Comment by jmerc77** [comment]
Agreed. I was trying to be quick an to the point. Also, torrent is banned
in some countries (i think) and some isps charge a fee or even block it all
together in the states. Definitely check your law books.

**Comment by Emilprivate** [comment, marked as answer]
Perhaps take a look here:
[updates-to-our-policies-regarding-exploits-malware-and-vulnerability-research](https://github.blog/2021-06-04-updates-to-our-policies-regarding-exploits-malware-and-vulnerability-research/)

That blog was from 2021, so I'm unsure if they've changed any policies since.

_"We explicitly permit dual-use security technologies and content related to research into vulnerabilities, malware, and exploits. We understand that many security research projects on GitHub are dual-use and broadly beneficial to the security community. We assume positive intention and use of these projects to promote and drive improvements across the ecosystem. This change modifies previously broad language that could be misinterpreted as hostile toward projects with dual-use, clarifying that such projects are welcome."_

**Reply by mmd18cury** [reply]
Great answer, 99,9% percent accuracy.
